js/ui/file_manager/ui.file_manager.toolbar.js

```diff
@@ -197,20 +157,51 @@ class FileManagerToolbar extends Widget {
         });
     }
 
-    _getItemConfigByCommandName(commandName) {
+    _configureItemByCommandName(commandName, item) {
+        let result = {};
+
         const command = this._commandManager.getCommandByName(commandName);
         if(command) {
-            return this._createCommandItem(command);
+            result = this._createCommandItem(command);
         }
 
         switch(commandName) {
             case "separator":
-                return this._createSeparatorItem();
+                result = this._createSeparatorItem();
+                break;
             case "viewMode":
-                return this._createViewModeItem();
+                result = this._createViewModeItem();
+                break;
         }
 
-        return {};
+        if(this._isDefaultItem(commandName)) {
+            const defaultConfig = DEFAULT_ITEM_CONFIGS[commandName];
+            extend(result, defaultConfig);
+            extend(result, item.visibilityMode ? { visibilityMode: item.visibilityMode } : {});
+            if(result.visibilityMode) {
+                extend(result, item.visible ? { visible: item.visible } : {});
+                extend(result, item.disabled ? { disabled: item.disabled } : {});
+            }
+            extend(result, item.location ? { location: item.location } : {});
+            extend(result, item.locateInMenu ? { locateInMenu: item.locateInMenu } : {});
+            extend(result.options, item.text ? { text: item.text } : {});
+            extend(result.options, item.icon ? { icon: item.icon } : {});
+        } else {
+            extend(result, item);
+            if(!result.widget) {
+                result.widget = "dxButton";
+            }
+        }
+
+        if(!result.commandName) {
```

Comment:
In case of the custom toolbar item we shouldn't  create `commandName` attribute with `undefined` value. We can change this code as below:
`if(commandName && !result.commandName)`